internal/work/dispatcher.go

```diff
@@ -340,3 +321,42 @@ func (d *Dispatcher) senderName(sender dbus.Sender) (string, error) {
 
 	return "", fmt.Errorf("cannot get name for sender: %v", sender)
 }
+
+// workerEventFromSignal creates an ipc.WorkerEvent from a DBus signal.
+func workerEventFromSignal(s *dbus.Signal) (*ipc.WorkerEvent, error) {
+	if s.Name != "com.redhat.Yggdrasil1.Worker1.Event" {
+		return nil, errIncorrectSignal
+	}
+
+	if len(s.Body) < 2 {
+		return nil, errMissingRequiredParameters
+	}
+
+	event := ipc.WorkerEvent{}
+
+	if len(s.Body) > 0 {
```

Comment:
I think that we can remove these checks for the length of `s.Body`, because we have already checked it

Reply:
I was trying to find a pattern that is recognizable and that clearly indicates which indices of the body slice will be accessed. I know the checks are superfluous, but they serve as sentinels to indicate that certain length slices are going to be used. I might be able to rework this into a switch using fallthrough cases, but I'm not convinced that would more clearly express my intention.

Reply:
Hint: you can simply use comments in code ;-).